File: benchmarks/dx/claude_agent_sdk/hello_world.py
# SPDX-License-Identifier: AGPL-3.0-only
"""Claude Agent SDK — minimal Hello World agent.

Net LOC: 7
Requires: pip install anthropic
NOTE: Requires ANTHROPIC_API_KEY in environment.

The Anthropic SDK gives the most direct path to the Claude API.
There is no "agent" abstraction — you call the messages API directly.
This is simultaneously the simplest and most explicit option.
"""
import anthropic

client = anthropic.Anthropic()

response = client.messages.create(
    model="claude-sonnet-4-5",
    max_tokens=256,
    system="You are a helpful assistant.",
    messages=[{"role": "user", "content": "Hello, what can you do?"}],
)

print(response.content[0].text)

# --- LOC count (net, no blanks/comments) ---
# import anthropic                                            1
# client = anthropic.Anthropic()                             1
# response = client.messages.create(                         1
#     model="claude-sonnet-4-5",                             1
#     max_tokens=256,                                        1
#     system="You are a helpful assistant.",                  1
#     messages=[{"role": "user", "content": "..."}],         1
# )                                                          1
# print(response.content[0].text)                            1
# TOTAL: 9 net LOC
#
# Honest note: this is a raw API call, not an "agent".
# The Claude Agent SDK does not provide high-level agent abstractions
# (AgentRunner, tool registry, session management). Those are exactly
# what Argentor provides on top of this baseline.

File: benchmarks/dx/argentor/hello_world.rs
//! Argentor — minimal "Hello World" agent.
//!
//! Time-to-first-agent: 15 net LOC (excluding blank lines / comments).
//! Boilerplate categories: ModelConfig (1), AgentRunner::new (1), run (1).
//!
//! Compile: add argentor-agent to Cargo.toml, then `cargo run`.
//! NOTE: Requires ANTHROPIC_API_KEY in environment.

use argentor_agent::{AgentRunner, ModelConfig};

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    let config = ModelConfig {
        model: "claude-sonnet-4".to_string(),
        max_tokens: 256,
        ..Default::default()
    };

    let mut agent = AgentRunner::new(config)?;

    let response = agent
        .run("You are a helpful assistant.", "Hello, what can you do?")
        .await?;

    println!("{}", response.content);
    Ok(())
}

// --- LOC count (net, no blanks/comments) ---
// use argentor_agent::{AgentRunner, ModelConfig};          1
// async fn main() -> anyhow::Result<()> {                  1
//     let config = ModelConfig {                            1
//         model: "claude-sonnet-4".to_string(),             1
//         max_tokens: 256,                                  1
//         ..Default::default()                              1
//     };                                                    1
//     let mut agent = AgentRunner::new(config)?;            1
//     let response = agent                                  1
//         .run("You are a helpful assistant.", "Hello...")  1
//         .await?;                                          1
//     println!("{}", response.content);                     1
//     Ok(())                                                1
// }                                                         1
// TOTAL: 14 net LOC
